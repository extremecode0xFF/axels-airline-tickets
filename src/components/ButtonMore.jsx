import React from 'react';
import { Button, Container } from 'react-bootstrap';
import styled from 'styled-components';

const Styles = styled(Container)`
  .btn {
    transition: all 0.2s ease-out;
    line-height: 50px;
    background-color: #2196f3;
    :focus {
      outline: none;
      box-shadow: none;
    }
    :hover {
      background-color: #0186f3;
      border: 1px solid black;
    }
  }
`;

export default function ButtonMore() {
  return (
    <Styles>
      <Button className="w-100" size="lg">
        Показать еще пять билетов!
      </Button>
    </Styles>
  );
}
