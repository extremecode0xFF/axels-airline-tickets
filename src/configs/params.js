const configFilterCheckbox = [
  {
    title: 'Без пересадок',
    query: 'transfer0',
  },
  {
    title: '1 пересадка',
    query: 'transfer1',
  },
  {
    title: '2 пересадки',
    query: 'transfer2',
  },
  {
    title: '3 пересадки',
    query: 'transfer3',
  },
];

const configFilterTabs = [
  {
    title: 'Самый дешевый',
    query: 'cheapest',
  },
  {
    title: 'Самый быстрый',
    query: 'fastest',
  },
  {
    title: 'Оптимальный',
    query: 'optimal',
  },
];

export { configFilterCheckbox, configFilterTabs };
