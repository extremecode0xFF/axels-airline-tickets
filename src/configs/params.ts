import { QuerySort } from './../types/queries';
import {
  TRANSFER_0,
  TRANSFER_1,
  TRANSFER_2,
  TRANSFER_3,
  CHEAPEST,
  FASTEST,
  OPTIMAL,
} from '../data/constants/queries';
import { ConfigParams } from '../types/config';
import { QueriesFilter } from '../types/queries';

const configFilterCheckbox: ConfigParams<QueriesFilter>[] = [
  {
    title: 'Без пересадок',
    query: TRANSFER_0,
  },
  {
    title: '1 пересадка',
    query: TRANSFER_1,
  },
  {
    title: '2 пересадки',
    query: TRANSFER_2,
  },
  {
    title: '3 пересадки',
    query: TRANSFER_3,
  },
];

const configFilterTabs: ConfigParams<QuerySort>[] = [
  {
    title: 'Самый дешевый',
    query: CHEAPEST,
  },
  {
    title: 'Самый быстрый',
    query: FASTEST,
  },
  {
    title: 'Оптимальный',
    query: OPTIMAL,
  },
];

export { configFilterCheckbox, configFilterTabs };
