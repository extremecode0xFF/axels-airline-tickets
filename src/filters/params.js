import { filterTicketsByStops } from '../assets/helpers/filterTickets';

const configFilterCheckbox = [
  {
    title: 'Все',
    action: (tickets) => tickets,
    query: { all: 'active' },
  },
  {
    title: 'Без пересадок',
    action: filterTicketsByStops(0),
    query: { withoutTransfer: 'active' },
  },
  {
    title: '1 пересадка',
    action: filterTicketsByStops(1),
    query: { transfer1: 'active' },
  },
  {
    title: '2 пересадки',
    action: filterTicketsByStops(2),
    query: { transfer2: 'active' },
  },
  {
    title: '3 пересадки',
    action: filterTicketsByStops(3),
    query: { transfer3: 'active' },
  },
];

const configFilterTabs = [
  {
    title: 'Самый дешевый',
    action: () => {},
    query: { fastest: 'active' },
  },
  {
    title: 'Самый быстрый',
    action: () => {},
    query: { cheapest: 'active' },
  },
  {
    title: 'Оптимальный',
    action: () => {},
    query: { optimal: 'active' },
  },
];

export { configFilterCheckbox, configFilterTabs };
