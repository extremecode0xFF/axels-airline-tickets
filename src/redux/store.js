import { configureStore } from '@reduxjs/toolkit';
import createSagaMiddleware from '@redux-saga/core';

import reducer from './ducks/tickets';
import { watcherSaga } from './ducks/rootSaga';

const sagaMiddleware = createSagaMiddleware();

const store = configureStore({
  reducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(sagaMiddleware),
});

sagaMiddleware.run(watcherSaga);

export default store;
