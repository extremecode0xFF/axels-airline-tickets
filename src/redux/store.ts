import { configureStore } from '@reduxjs/toolkit';
import createSagaMiddleware from '@redux-saga/core';
import { all } from 'redux-saga/effects';

import reducer from './ducks/tickets';
import {
  watcherGetTickets,
  watcherSetFilteredTickets,
  watcherSetSortedTickets,
} from './ducks/tickets';

const sagaMiddleware = createSagaMiddleware();

const store = configureStore({
  reducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(sagaMiddleware),
});

function* watcherSaga() {
  yield all([
    watcherGetTickets(),
    watcherSetFilteredTickets(),
    watcherSetSortedTickets(),
  ]);
}

sagaMiddleware.run(watcherSaga);

export type RootState = ReturnType<typeof store.getState>;

export default store;
