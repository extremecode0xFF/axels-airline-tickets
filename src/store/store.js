import { configureStore } from '@reduxjs/toolkit';
import createSagaMiddleware from '@redux-saga/core';
import { watcherSaga } from './sagas/rootSaga';
import reducer from './ducks/tickets';

const sagaMiddleware = createSagaMiddleware();

const store = configureStore({
  reducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(sagaMiddleware),
});

sagaMiddleware.run(watcherSaga);

export { store };
