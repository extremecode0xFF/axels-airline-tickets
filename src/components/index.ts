import ButtonMore from './ButtonMore';
import ButtonTabs from './ButtonTabs';
import Filters from './Filters';
import Header from './Header';
import Ticket from './Ticket';
import TicketList from './TicketList';

export { ButtonMore, ButtonTabs, Filters, Header, Ticket, TicketList };
