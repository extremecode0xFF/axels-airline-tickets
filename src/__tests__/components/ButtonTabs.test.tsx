import React from 'react';
import renderer from 'react-test-renderer';

import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { ButtonTabs } from '../../components';

import { configFilterTabs } from '../../configs/params';
import store from '../../redux/store';

describe('ButtonTabs', () => {
  test('match snapshot', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Provider store={store}>
            <ButtonTabs config={configFilterTabs} />
          </Provider>
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
