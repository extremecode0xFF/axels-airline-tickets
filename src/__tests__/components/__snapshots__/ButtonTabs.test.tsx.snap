// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonTabs match snapshot 1`] = `
<div
  className="sc-gsnTZi iPbkSZ mb-3 container"
>
  <div
    className="w-100 btn-group btn-group-lg"
    role="group"
  >
    <button
      className="text-uppercase btn btn-light"
      disabled={false}
      id="cheapest"
      name="radio"
      onClick={[Function]}
      type="button"
      value="cheapest"
    >
      Самый дешевый
    </button>
    <button
      className="text-uppercase btn btn-light"
      disabled={false}
      id="fastest"
      name="radio"
      onClick={[Function]}
      type="button"
      value="fastest"
    >
      Самый быстрый
    </button>
    <button
      className="text-uppercase btn btn-light"
      disabled={false}
      id="optimal"
      name="radio"
      onClick={[Function]}
      type="button"
      value="optimal"
    >
      Оптимальный
    </button>
  </div>
</div>
`;
